pkg/command/reconciler.go

```diff
@@ -180,21 +195,45 @@ func (r *Reconciliation) initChildJobDetails() {
 	}
 	if !found {
 		klog.V(1).Infof(
-			"Job status.active is not set: Kind %q: Job %q / %q",
+			"Job status.conditions is not set: Kind %q: Job %q / %q",
 			r.childJob.GetKind(),
 			r.childJob.GetNamespace(),
 			r.childJob.GetName(),
 		)
-		// Job's status.active is not set
+		// Job's status.conditions is not set
 		//
 		// Nothing to do
 		// Wait for next reconcile
 		return
 	}
-
-	if activeCount > 0 {
-		r.isChildJobCompleted = true
+	// Look for condition type complete
+	// if found then mark isChildJobCompleted as true
+	for _, value := range jobConditions {
+		condition, ok := value.(map[string]interface{})
+		if !ok {
+			r.err = errors.Errorf(
+				"Job status.condition is not map[string]interface{} got %T: "+
+					"kind %q: Job %q / %q",
+				value,
+				r.childJob.GetKind(),
+				r.childJob.GetNamespace(),
+				r.childJob.GetName(),
+			)
+			return
+		}
+		condType := condition["type"].(string)
+		if condType == types.JobPhaseCompleted {
+			condStatus := condition["status"].(string)
+			if condStatus == "True" {
+				r.isChildJobCompleted = true
```

Comment:
As per this logic if any condition status is True we mark the job as completed.
Is that okay?

Reply:
No, we are marking the job as completed only when the condition type is [Complete](https://github.com/kubernetes/kubernetes/blob/b1a2a99f1fe1622d20b68abaa99348e19a35c704/pkg/apis/batch/types.go#L194) and status is set to `True`